client/python/openlineage/client/transport/file.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2018-2023 contributors to the Marquez project
+# SPDX-License-Identifier: Apache-2.0
+from openlineage.client.run import RunEvent
+from openlineage.client.serde import Serde
+
+from dataclasses import dataclass
+from datetime import datetime
+
+@dataclass
+class FileConfig:
+    log_file_path: str
+    append: bool = False
+
+    @classmethod
+    def from_dict(cls, params: dict):
+        if 'log_file_path' not in params:
+            raise RuntimeError("`log_file_path` key not passed to FileConfig")
+
+        log_file_path = params['log_file_path']
+
+        log_file_handle = open(log_file_path, "a" if cls.append else "w")
+        if not log_file_handle.writable():
+            raise RuntimeError(f"Log file {log_file_path} is not writeable")
+
+        return cls(log_file_path)
+
+
+class FileTransport:
+    kind = "file"
+    config = FileConfig
+
+    def __init__(self, config: FileConfig) -> None:
+        self.config = config
+
+    def emit(self, event: RunEvent):
+
+        if self.config.append:
+          log_file_path = self.config.log_file_path
+        else:
+          timestr = datetime.now().strftime("%Y%m%d-%H%M%S.%f")
+          log_file_path = f"{self.config.log_file_path}-{timestr}"
+
+        with open(log_file_path, "a" if self.config.append else "w") as log_file_handle:
+            log_file_handle.write(Serde.to_json(event) + "\n")
+
+class FileTransportFactory:
```

Comment:
This is not needed. `DefaultTransportFactory` will be able to instantiate the transport.

Reply:
thanks!  I deleted TransportFactory, run black & add header on test on my last commit

Reply:
signing done